### ISSUE TITLE
mod_http_offline not working for ejabberd 15.11

### ISSUE THREAD

**Issue by twais** [processone/ejabberd#930]
I have recently upgraded to ejabberd version 15.11. All modules are working apart from the mod_http_offline. I am able to successfully compile the erl which generates a beam file, which I copy to  the ebin folder and also included the module on my ejabberd.yml folder (mod_http_offline: []). I am however getting an error on restart as below

```
09:55:58.141 [critical] Problem starting the module mod_http_offline for host <<"elmajabber">> 
 options: []
 error: undef
[{p1_logger,info_msg,[mod_http_offline,22,"mod_http_offline loading",[]],[]},
 {mod_http_offline,start,2,
                   [{file,"Downloads/mod_http_offline.erl"},{line,22}]},
 {gen_mod,start_module,3,[{file,"src/gen_mod.erl"},{line,98}]},
 {lists,foreach,2,[{file,"lists.erl"},{line,1336}]},
 {ejabberd_app,start,2,[{file,"src/ejabberd_app.erl"},{line,74}]},
 {application_master,start_it_old,4,
                     [{file,"application_master.erl"},{line,272}]}]
09:55:58.142 [critical] ejabberd initialization was aborted because a module start failed.

Crash dump was written to: //var/log/ejabberd/erl_crash_20160128-095554.dump
Problem starting the module mod_http_offline for host <<"elmajabber">> 
 options: []
 error: undef
[{p1_logger,info_msg,[mod_http_offline,22,"mod_http_offline loading",[]],[]},
 {mod_http_offline,sta
```
**Below is my module**. Please assist..

```
%%start

%% name of module must match file name
-module(mod_http_offline).
 
-author("Earl The Squirrel").
 
%% Every ejabberd module implements the gen_mod behavior
%% The gen_mod behavior requires two functions: start/2 and stop/1
-behaviour(gen_mod).
 
%% public methods for this module
-export([start/2, stop/1, create_message/3]).
 
%% included for writing to ejabberd log file
-include("ejabberd.hrl").
 
%% ejabberd functions for JID manipulation called jlib.
-include("jlib.hrl").

-include("logger.hrl").
 
start(_Host, _Opt) ->
        ?INFO_MSG("mod_http_offline loading", []),
        inets:start(),
        ?INFO_MSG("HTTP client started", []),
        post_offline_message("testFrom", "testTo", "testBody"),
        ejabberd_hooks:add(offline_message_hook, _Host, ?MODULE, create_message, 50).  
  
stop (_Host) ->
        ?INFO_MSG("stopping mod_http_offline", []),
        ejabberd_hooks:delete(offline_message_hook, _Host, ?MODULE, create_message, 50).

create_message(_From, _To, Packet) ->
        %%Type = xml:get_tag_attr_s("type", Packet),
        FromS = xml:get_path_s(Packet, [{elem, "FromUser"}, cdata]),
        ToS = xml:get_tag_attr_s("to", Packet),
        Body = xml:get_path_s(Packet, [{elem, "body"}, cdata]),
        post_offline_message(FromS, ToS, Body).
 
post_offline_message(From, To, Body) ->
         ?INFO_MSG("Posting From ~p To ~p Body ~p~n",[From, To, Body]),
         httpc:request(post, {"http://172.10.3.21/chat/offline/offlinemsg.php",[],
         "application/x-www-form-urlencoded",
         lists:concat(["From=", From,"&To=", To,"&Body=", Body])}, [], []),
         ?INFO_MSG("post request sent", []).
%%end
```

**Comment by mremond** [comment]
undef for p1_logger means that this module is missing. mod_http_offline crash means it was compiled with incorrect Lager define. It is true in your case for ejabberd, but you compiled mod_http_offline without this define.

Moreover, I will close the ticket as it seems more relevant for ejabberd-contrib project.

**Comment by shajunr88** [comment]
mod_http_offline is available in modules_available ejabberd 16.09.But how to give url details in order to call particular web page.
